include/oneapi/dpl/pstl/iterator_impl.h

```diff
@@ -91,13 +91,9 @@ class zip_forward_iterator
 
     zip_forward_iterator() : __my_it_() {}
     explicit zip_forward_iterator(_Types... __args) : __my_it_(::std::make_tuple(__args...)) {}
-    zip_forward_iterator(const zip_forward_iterator& __input) : __my_it_(__input.__my_it_) {}
+    zip_forward_iterator(const zip_forward_iterator& __input) = default;
```

Comment:
I would propose to remove the definition of copy constructor and copy assignment completely. Or add definitions of move assignment and move constructor. Please play with this https://godbolt.org/z/M19dbPhMP and let me know if you have questions.

Reply:
Probably also better to rewrite default constructor as
`zip_forward_iterator() = default;`

Reply:
Thanks for this example.  It is helpful.

Reply:
Probably if we apply https://github.com/oneapi-src/oneDPL/pull/1374 it's wouldn't required anymore at all.